### DISCUSSION TITLE
DevTools panel direction in RTL pages

### DISCUSSION THREAD

**Discussion by m0bar3z** [TanStack/query#7563] [Ideas]
When I install DevTools in my project where the page direction is set to RTL (right-to-left), the DevTools panel order gets messed up, making it hard to read and use. Wouldn't it be better to force the DevTools panel to stay in LTR (left-to-right) regardless of the page direction?

![ss1](https://github.com/TanStack/query/assets/39567859/068a1e8e-95f1-48bf-8171-5df1cd5b7abb)
![ss2](https://github.com/TanStack/query/assets/39567859/a5a9a276-071f-4095-92ac-918b471347a5)
![ss3](https://github.com/TanStack/query/assets/39567859/e195f277-c422-4899-8e74-841ce1359583)

**Comment by aminlotfi** [comment]
I encountered the same issue with the React Query DevTools panel in RTL pages, where the panel's layout becomes confusing and hard to use.  

As a temporary fix in my project, I wrapped the DevTools component in a `<div>` with `dir="ltr"` like this:  

```jsx
<QueryClientProvider client={queryClient}>
      {children}
      {process.env.NODE_ENV === 'development' && (
        <div dir='ltr'>
          <ReactQueryDevtools initialIsOpen={false} />
        </div>
      )}
</QueryClientProvider>
```

This forces the DevTools panel to render in LTR, regardless of the page's direction, and resolves the issue for now.  

I believe a more permanent fix in the library itself would be beneficial. I’d also love to contribute to addressing this issue. Could someone guide me if there are any preferences or specific constraints for implementing this?

**Reply by TkDodo** [reply]
if adding `dir='ltr'` to the top level element that renders the ReactQueryDevtools fixes the problem, please open a PR 🙏
